src/full/Agda/TypeChecking/Conversion.hs

```diff
@@ -1791,17 +1807,26 @@ equalSort s1 s2 = do
           -- @SSetω(n+1)@ is the successor sort of exactly @SSetω(n)@.
           -- @SSetω@ is the successor sort of exactly @SSetω@ if
           -- --type-in-type or --omega-in-omega is enabled.
-          -- The same is only true for @Setωᵢ@ if @Propω...@ are disabled.
-          Inf u n | invertibleSort propEnabled u ->
-            if n > 0 then equalSort (Inf u $ n - 1) s2 else do
-              infInInf <- (optOmegaInOmega <$> pragmaOptions) `or2M` typeInType
-              if | infInInf  -> equalSort s1 s2
-                 | otherwise -> no
-          -- @Prop l@, @SizeUniv@ and @LevelUniv@ are not successor sorts
-          Prop{}      -> no
-          Inf UProp _ -> no
-          SizeUniv{}  -> no
-          LevelUniv{} -> no
+          -- The same is only true for @Setω(n+1)@ if @Propω...@ are disabled.
+          -- @Setω@ is the successor sort of @Setω@ (type:type) or @SizeUniv@ (--sized-types).
+          Inf u 0 -> do
+              -- Compute the predecessor(s) of (S)Setω and return it if it is unique.
+              sizedTypesEnabled <- sizedTypesOption
+              -- guardedEnabled <- optGuarded <$> pragmaOptions
+              case concat
+                [ [ s1       | u /= UProp, infInInf ]
+                , [ dummy    | u == UType, infInInf, propEnabled, let dummy = Inf UProp 0 ]
+                    -- We enter a dummy into the solution set if --prop makes predecessor ambiguous.
+                , [ SizeUniv | u == UType, sizedTypesEnabled ]
+                -- , [ LockUniv | guardedEnabled ]  -- LockUniv is actually in Set₁, not Setω
+                ]
+                of
+                [ s ] -> equalSort s s2
+                []    -> no
+                _     -> postpone
```

Comment:
@jespercockx Maybe have a look at the `Inf u 0` case?  I was going back an forth until I arrived at this formulation...

Reply:
LGTM

Reply:
Thanks!